File: 3.无重复字符的最长子串.go
/*
 * @lc app=leetcode.cn id=3 lang=golang
 *
 * [3] 无重复字符的最长子串
 */

// @lc code=start
func lengthOfLongestSubstring(s string) int {
	if len(s) <= 1 {
		return len(s)
	}
    result := 0
	left := 0
	for right := 1; right < len(s); right++ {
		for i := left; i < right; i++ {
			if s[i] == s[right] {
				left = i+1
			}
		}
		result = max(right-left+1, result)
	}
	return result
}
// @lc code=end

File: 3.无重复字符的最长子串.py
#
# @lc app=leetcode.cn id=3 lang=python3
#
# [3] 无重复字符的最长子串
#

# @lc code=start
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if not s:
            return 0
        
        visited = set()

        l, r = 0,0
        visited.add(s[0])

        result = 1
        for r in range(1, len(s)):
            while s[r] in visited:
                visited.remove(s[l])
                l += 1
            visited.add(s[r])
            result = max(result, len(visited))
        
        return result
            
# @lc code=end

